argocd/structure_application_set.go

```diff
@@ -31,6 +31,10 @@ func expandApplicationSetSpec(d *schema.ResourceData, featureMultipleApplication
 
 	spec.GoTemplate = s["go_template"].(bool)
 
+	for _, k := range s["go_template_options"].([]interface{}) {
+		spec.GoTemplateOptions = append(spec.GoTemplateOptions, k.(string))
+	}
```

Comment:
```suggestion
	if v, ok := s["go_template_options"]; ok {
		opts := v.(*schema.Set).List()
		for _, opt := range opts {
			spec.GoTemplateOptions = append(spec.GoTemplateOptions, opt.(string))
		}
```

... change need due to change in `type`